functions/go/set-labels/transformer/labels.go

```diff
@@ -0,0 +1,137 @@
+package transformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sort"
+	"strings"
+)
+
+// the similar struct esxits in resid.GVK, but there is no function to create an GVK struct without using kyaml
+type FieldSpec struct {
+	Group              string `json:"group,omitempty" yaml:"group,omitempty"`
+	Version            string `json:"version,omitempty" yaml:"version,omitempty"`
+	Kind               string `json:"kind,omitempty" yaml:"kind,omitempty"`
+	Path               string `json:"path,omitempty" yaml:"path,omitempty"` // seperated by /
+	CreateIfNotPresent bool   `json:"create,omitempty" yaml:"create,omitempty"`
```

Comment:
Can you also add a TODO here about we will change this "createIfNotPresent" to be a configurable field?

Reply:
> we will change this "createIfNotPresent" to be a configurable field

@yuwenma Can you please explain this a bit?